capa/features/extractors/binexport2/__init__.py

```diff
@@ -235,6 +235,13 @@ def get_function_name_by_address(self, address: int) -> str:
         vertex_index: int = self.vertex_index_by_address[address]
         return self.get_function_name_by_vertex(vertex_index)
 
+    def get_instruction_by_address(self, address: int) -> BinExport2.Instruction:
+        for i, be2_insn in enumerate(self.be2.instruction):
+            insn = self.get_insn_address(i)
+            if address == insn:
+                return be2_insn
```

Comment:
this looks very slow. lets build an index first and use that instead (if there isn't one already??).

Reply:
`self.insn_address_by_index`

Reply:
right, we need `insn_by_address` for this

Reply:
Following, as @williballenthin mentioned, let's use an index here, but as you pointed out @mr-tz we don't have instructions indexed by address, only by BinExport instruction index. Let's extend https://github.com/mandiant/capa/blob/c662176e38920b741dccca50637a664e9c94664d/capa/features/extractors/binexport2/__init__.py#L172-L188 to additionally populate an instruction-by-address index.

Reply:
ah sorry i missed the direction of the lookup

Reply:
bf38f225